garden_ai/client.py

```diff
@@ -253,13 +258,17 @@ def create_pipeline(
             data["authors"] = authors
         if title:
             data["title"] = title
+        data["doi"] = data.get("doi") or "placeholder"
 
         pipeline = Pipeline(**data)
-        record = local_data.get_local_pipeline_by_uuid(pipeline.uuid)
+        record = local_data.get_local_pipeline_by_doi(pipeline.doi)
```

Comment:
we won't need this check to reuse a DOI anymore, now that DOIs are the only thing we're indexing records on.

On that note, there's another higher-level design smell here that I think is worth pointing out -- if we're indexing our data on a single field, we really don't want to code around it in such a way that collisions are possible for that field, which is very much the case when introducing non-unique placeholder/default values for the index.

Reply:
My favorite example of this is the guy in California who registered the license plate "NULL", thinking he'd be impossible to ticket, but instead ended up getting hit every single time a traffic cop typo-d/skipped the license plate field writing the ticket. Here, it looks like the worst that happens is a redundant assignment, but we don't want our future selves to have to worry about this kind of thing if we can help it.